### PR TITLE
Add Github Actions CI support

### PR DESCRIPTION
**Pull request by guilledk**
I still have to do some more testing to fix some possible flakey tests, this also adds testing on mac!

### CODE REVIEW

**Comment by guilledk** [comment]
Ok, pretty sure flakey tests are fixed, to see the Github Actions jobs for this commit: https://github.com/guilledk/tractor/runs/1062578871

I guess once you merge the actions will start appearing in this repo and not my fork (?

File: tests/conftest.py [hunk 1]
```diff
@@ -74,11 +77,21 @@ def spawn_backend(request):
...
+
 @pytest.fixture(scope='session')
-def travis():
-    """Bool determining whether running inside TravisCI.
+def ci_env() -> Optional[CIEnvoirment]:
+    """Detect CI envoirment.
     """
-    return os.environ.get('TRAVIS', False)
+    if os.environ.get('TRAVIS'):
+        return CIEnvoirment.Travis
```
Comment (goodboy) [inline tests/conftest.py:90]: Hmm, so looking at the [logic below](https://github.com/goodboy/tractor/pull/152/files#diff-d3eea2f9e263e7cd8e79cc7c81011ceaR208) it seems you really are just setting a `bool` yah?

Do we really need an `Enum` type for this?

**Comment by guilledk** [comment]
Ok, the current testing suite is the equivalent of the ``.travis.yml``, I'll continue the `Darwin` integration in another branch

**Review by goodboy — APPROVED** [review]

File: tests/conftest.py (hunk 1 above)
Comment (pquentin) [reply tests/conftest.py]: Did you mean "environment" instead?

Comment (goodboy) [reply tests/conftest.py]: @pquentin this code got replaced I'm pretty sure ;)

**Pull request merged**